src/main/java/de/tum/in/www1/artemis/service/NotificationSettingsService.java

```diff
@@ -0,0 +1,85 @@
+            "notification.lecture-notification.new-post-for-lecture", new NotificationType[] { NEW_POST_FOR_LECTURE },
+            "notification.lecture-notification.new-answer-post-for-lecture", new NotificationType[] { NEW_ANSWER_POST_FOR_LECTURE },
+            "notification.instructor-exclusive-notification.course-and-exam-archiving-started", new NotificationType[] { EXAM_ARCHIVE_STARTED, COURSE_ARCHIVE_STARTED });
+
+    /**
+     * Finds the deactivated NotificationTypes based on the user's NotificationOptions
+     * @param notificationOptions which should be mapped to their respective NotificationTypes and filtered by activation status
+     * @return a set of NotificationTypes which are deactivated by the current user's notification settings
+     */
+    public Set<NotificationType> findDeactivatedNotificationTypes(Set<NotificationOption> notificationOptions) {
+        Map<NotificationType, Boolean> notificationOptionWitchActivationStatusMap = convertNotificationOptionsToNotificationTypesWithActivationStatus(notificationOptions);
+        Set<NotificationType> deactivatedNotificationTypes = new HashSet<>();
+        notificationOptionWitchActivationStatusMap.forEach((notificationType, isActivated) -> {
+            if (!isActivated) {
+                deactivatedNotificationTypes.add(notificationType);
+            }
+        });
+        return deactivatedNotificationTypes;
+    }
+
+    /**
+     * Converts the provided NotificationType Set to a String Set (representing the titles from NotificationTitleTypeConstants)
+     * @param types Set that should be converted to String
+     * @return the converted String Set
+     */
+    public Set<String> convertNotificationTypesToTitles(Set<NotificationType> types) {
+        return types.stream().map(NotificationTitleTypeConstants::findCorrespondingNotificationTitle).collect(Collectors.toSet());
+    }
+
+    /**
+     * Converts the provided NotificationOptions to a map of corresponding NotificationTypes and activation status.
+     * @param notificationOptions which will be mapped to their respective NotificationTypes with respect to their activation status
+     * @return a map with key of NotificationType and value Boolean indicating which types are (de)activated by the user's notification settings
+     */
+    private Map<NotificationType, Boolean> convertNotificationOptionsToNotificationTypesWithActivationStatus(Set<NotificationOption> notificationOptions) {
+        Map<NotificationType, Boolean> resultingMap = new HashMap<>();
+        for (NotificationOption option : notificationOptions) {
+            NotificationType[] tmpNotificationTypes = NOTIFICATION_OPTION_SPECIFIER_TO_NOTIFICATION_TYPES_MAP.get(option.getOptionSpecifier());
+            if (tmpNotificationTypes != null) {
+                for (NotificationType type : tmpNotificationTypes) {
+                    resultingMap.put(type, option.isWebapp());
+                }
+            }
```

Comment:
```suggestion
            NotificationType[] tmpNotificationTypes = NOTIFICATION_OPTION_SPECIFIER_TO_NOTIFICATION_TYPES_MAP.getOrDefault(option.getOptionSpecifier(), new NotificationType[0]);
            for (NotificationType type : tmpNotificationTypes) {
                resultingMap.put(type, option.isWebapp());
            }
```

I think this could also be improved. The for loop won't loop if list is empty, no need for null check.